addons/web/static/tests/views/fields/boolean_favorite_field_tests.js

```diff
@@ -74,6 +74,86 @@ QUnit.module("Fields", (hooks) => {
         );
     });
 
+    QUnit.test("FavoriteField saves changes by default", async function (assert) {
+        await makeView({
+            type: "kanban",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <kanban>
+                    <templates>
+                        <t t-name="kanban-box">
+                            <div>
+                                <field name="bar" widget="boolean_favorite" />
+                            </div>
+                        </t>
+                    </templates>
+                </kanban>`,
+            mockRPC(route, args) {
+                if (args.method === "write" && args.model === "partner") {
+                    assert.step("save");
+                    assert.deepEqual(args.args, [[1], { bar: false }]);
+                }
+            },
+            domain: [["id", "=", 1]],
+        });
+
+        // click on favorite
+        await click(target, ".o_field_widget .o_favorite");
+        assert.containsNone(
+            target,
+            ".o_kanban_record  .o_field_widget .o_favorite > a i.fa.fa-star",
+            "should not be favorite"
+        );
+        assert.strictEqual(
+            target.querySelector(".o_kanban_record .o_field_widget .o_favorite > a").textContent,
+            " Add to Favorites",
+            'the label should say "Add to Favorites"'
+        );
+
+        assert.verifySteps(["save"]);
+    });
+
+    QUnit.test("FavoriteField does not save changes by default", async function (assert) {
```

Comment:
sorry but the test name makes no sense (even more with respect to the previous test). By default, it saves. But this can be disabled with an option. So this one should be something like "FavoriteField doesn't save if autosave option is set to false"